src/push_service/cdn.rs

```diff
@@ -417,18 +414,25 @@ impl PushService {
 
         trace!("digested content");
 
-        self.request(
+        let mut request = self.request(
             Method::PATCH,
             Endpoint::cdn(3, resumable_url.path()),
             HttpAuthOverride::Unidentified,
-        )?
-        .header("Tus-Resumable", "1.0.0")
-        .header("Upload-Offset", resume_info.content_start)
-        .header("Upload-Length", buf.len())
-        .header(CONTENT_TYPE, content_type)
-        .body(buf)
-        .send()
-        .await?;
+        )?;
+
+        for (key, value) in headers {
+            request = request.header(key, value);
+        }
```

Comment:
Possibly better to use [`headers`](https://docs.rs/reqwest/latest/reqwest/struct.RequestBuilder.html#method.headers) here.